packages/frontend/src/components/common/Filters/FilterInputs/DefaultFilterInputs.tsx

```diff
@@ -91,28 +91,31 @@ const DefaultFilterInputs = <T extends ConditionalRule>({
         case FilterOperator.IN_THE_PAST:
         case FilterOperator.IN_THE_NEXT:
         case FilterOperator.IN_THE_CURRENT:
+        case FilterOperator.IN_BETWEEN:
             const parsedValue = parseInt(rule.values?.[0] as string, 10);
             return (
                 <NumericInput
                     className={disabled ? 'disabled-filter' : ''}
                     disabled={disabled}
                     fill
-                    value={isNaN(parsedValue) ? undefined : parsedValue}
-                    min={0}
-                    onValueChange={(value) =>
+                    value={isNaN(parsedValue) ? '' : parsedValue}
+                    onValueChange={(value) => {
+                        const normalizedValue = isNaN(value)
+                            ? undefined
+                            : value;
+
                         onChange({
                             ...rule,
-                            values: [value],
-                        })
-                    }
+                            values: [normalizedValue],
+                        });
+                    }}
                 />
             );
-        default: {
-            const never: never = rule.operator;
-            throw Error(
-                `No form implemented for String filter operator ${filterOperator}`,
+        default:
+            return assertUnreachable(
+                rule.operator,
```

Comment:
this was silently failing because we don't have TSC for frontend during the build